src/docs/cookbook/animation/physics-simulation.md

```diff
@@ -0,0 +1,393 @@
+dragged to, to the point in the center .
+
+```dart
+Animation<Alignment> _animation;
+
+void _updateAnimation() {
+ _animation = _controller.drive(
+   AlignmentTween(
+     begin: _dragAlignment,
+     end: Alignment.center,
+   ),
+ );
+}
+```
+
+Call `_updateAnimation` in `initState`. This gives `_animation` an initial
+value. Then update `_dragAlignment` when the `AnimationController` produces a
+value:
+
+```dart
+@override
+void initState() {
+  _controller = AnimationController(vsync: this, duration: Duration(seconds: 1));
+  _controller.addListener(() {
+    setState(() {
+      _dragAlignment = _animation.value;
+    });
+  });
+
+  _updateAnimation();
+  super.initState();
+}
+
+```
+
+Next, make the `Align` widget use the `_dragAlignment` field:
+
+```dart
+child: Align(
+  alignment: _dragAlignment,
+  child: Card(
+    child: widget.child,
+  ),
+),
+```
+
+
+Finally, update the `GestureDetector` to manage the animation controller:
+
+
+```dart
+onPanDown: (details) {
+ _controller.stop();
+},
+onPanUpdate: (details) {
+ setState(() {
+   _dragAlignment += Alignment(
+     details.delta.dx / (size.width / 2),
+     details.delta.dy / (size.height / 2),
+   );
+ });
+},
+onPanEnd: (details) {
+ _updateAnimation();
+ _controller.reset();
+ _controller.forward();
+},
+```
+
+## Step 4: Calculate the velocity to simulate a springing motion
+
+The last step is to do a little math, to calculate the velocity of the widget
+after it's finished being dragged. This is so that the widget realistically
+continues at that speed before being snapped back. ( `_updateAnimation()`
+already sets the direction  by setting the animation's start and end alignment.)
+
+The `onPanEnd` callback provides a `DragEndDetails` object with the same
+velocity that the user was dragging their finger at. This value is in pixels per
+second, but the `Align` widget doesn't use pixels. It uses coordinate values
+between [-1.0, -1.0] and [1.0, 1.0] , where [0.0, 0.0] represents the center.
+The `size` calculated in step 2 is used to convert pixels to coordinate values
+in this range.
+
+Finally, `AnimationController` has an `animateWith()` method that can be given a
+`SpringSimulation`:
+
+```dart
+onPanEnd: (details) {
+  _updateAnimation();
+  // Calculate the velocity relative to the unit interval, [0,1],
+  // used by the animation controller.
+  var pxPerSecond = details.velocity.pixelsPerSecond;
+  var unitsPerSecondX = pxPerSecond.dx / size.width;
+  var unitsPerSecondY = pxPerSecond.dy / size.height;
+  var unitsPerSecond = Offset(unitsPerSecondX, unitsPerSecondY);
+  var unitVelocity = unitsPerSecond.distance;
+
+  var spring = SpringDescription(
+    mass: 30,
+    stiffness: 1,
+    damping: 1,
+  );
+  var simulation = SpringSimulation(spring, 0, 1, -unitVelocity);
+
+  _controller.animateWith(simulation);
+},
+```
+
+{{site.alert.note}}
+Now that the animation controller uses a simulation it's `duration` argument is
+no longer required.
```

Comment:
@goderbauer is there anything you feel should be explained here? Is this a best-practice?

Reply:
Intent the note content by 2 spaces per line.